examples/barotropic_tide.jl

```diff
@@ -0,0 +1,285 @@
+# # Barotropic tide
+#
+# In this example, we simulate the evolution of a barotropic tide over a hill.
+#
+# ## Install dependencies
+#
+# First let's make sure we have all required packages installed.
+
+# ```julia
+# using Pkg
+# pkg"add Oceananigans, CairoMakie"
+# ```
+
+using Oceananigans
+using Oceananigans.Units
+
+# ## Grid
+
+# We use an immersed boundary two-dimensional grid (in ``x``--``z``) that is is periodic in
+# the ``x``-direction. To construct an immersed boundary grid we first need to create what
+# we refer to as "underlying grid", which the grid that encompasses the immersed boundary.
+
+Nx, Nz = 200, 80
+
+H  = 2kilometers
+Lx = 2000kilometers
+
+underlying_grid = RectilinearGrid(size = (Nx, Nz),
+                                  x = (-Lx/2, Lx/2),
+                                  z = (-H, 0),
+                                  halo = (4, 4),
```

Comment:
It's too bad we need this. Should we make the default halo size 4?

Reply:
I think we should.
Is there a downside to that?
Memory footprint is really minimal, right?